packages/clerk-js/src/core/clerk.ts

```diff
@@ -720,6 +720,10 @@ export class Clerk implements ClerkInterface {
       eventBus.dispatch(events.TokenUpdate, { token: null });
     }
 
+    if (session?.lastActiveToken) {
+      eventBus.dispatch(events.TokenUpdate, { token: session.lastActiveToken });
+    }
+
```

Comment:
I don't think this will fix the issue because the `TokenUpdate` event is already triggered during session instance initialization when the `lastActiveToken` exists: https://github.com/clerk/javascript/blob/3c8fda28cfc99d6266397b75ebe1e6161192283c/packages/clerk-js/src/core/resources/Session.ts#L44

Reply:
I've updated the description, please take a look

Reply:
ok. it makes more sense now that this will fix the current issue.
Can we add some comments above this change that describes why we need this line of code?
Also instead of applying this change, why don't we introduce a separate event  `ClientUpdate` to handle the clientUat cookie when the `Clerk.client` is updated?

Reply:
Also let's make sure that there is a test to cover that the issue has been fixed before merging this change

Reply:
> Also instead of applying this change, why don't we introduce a separate event ClientUpdate to handle the clientUat cookie when the Clerk.client is updated?

I'd say that the current implementation needs a holistic refactor so minor tweaks won't make a difference at the momennt

Reply:
Added a test 👍🏻